py-polars/tests/unit/io/test_iceberg.py

```diff
@@ -163,3 +163,40 @@ def test_parse_lteq(self) -> None:
 
         expr = _to_ast("(pa.compute.field('ts') <= '2023-08-08')")
         assert _convert_predicate(expr) == LessThanOrEqual("ts", "2023-08-08")
+
+
+@pytest.mark.slow()
+@pytest.mark.write_disk()
+@pytest.mark.filterwarnings(
+    "ignore:No preferred file implementation for scheme*:UserWarning"
+)
+def test_write_iceberg(tmp_path: Path) -> None:
+    df = pl.DataFrame(
+        {
+            "foo": [1, 2, 3, 4, 5],
+            "bar": [6, 7, 8, 9, 10],
+            "ham": ["a", "b", "c", "d", "e"],
+        }
+    )
```

Comment:
This needs much more extensive testing. Try using the `df` fixture (it has a mix of many data types), or define your own dataframe with lots of different data types.

Reply:
Thanks! I'll take a look at this fixture